examples/plot_pitfalls_permutation_importance.py

```diff
@@ -0,0 +1,298 @@
+"""
+Pitfalls of Permutation Feature Importance (PFI)
+===============================================
+
+In this example, we illustrate the pitfalls of using permutation feature importance
+(PFI) on the California housing dataset.
+PFI measures the importance of a variable. However, it does not measure conditional
+importance and does not provide statistical control over the risk of making false
+discoveries, i.e., the risk of declaring a variable as important when it is not.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from matplotlib.lines import Line2D
+from scipy.stats import ttest_1samp
+from sklearn.base import clone
+from sklearn.compose import TransformedTargetRegressor
+from sklearn.datasets import fetch_california_housing
+from sklearn.linear_model import RidgeCV
+from sklearn.metrics import r2_score
+from sklearn.model_selection import KFold, train_test_split
+from sklearn.neural_network import MLPRegressor
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+
+from hidimstat import CPI, PermutationImportance
+from hidimstat.conditional_sampling import ConditionalSampler
+
+#############################################################################
+# Load the California housing dataset and add a spurious feature
+# ------------------------------------------------------------------
+# The California housing dataset is a regression dataset with 8 features. We add a
+# spurious feature that is a linear combination of 3 features plus some noise.
+# The spurious feature does not provide any additional information about the target.
+
+dataset = fetch_california_housing()
+X, y = dataset.data, dataset.target
+rng = np.random.RandomState(0)
+
+redundant_coef = rng.choice(np.arange(X.shape[1]), size=(3,), replace=False)
+X_spurious = X[:, redundant_coef].sum(axis=1)
+X_spurious += rng.normal(0, scale=np.std(X_spurious), size=X.shape[0])
+X = np.hstack([X, X_spurious[:, np.newaxis]])
+feature_names = dataset.feature_names + ["Spurious"]
+
+# Compute the correlation matrix
+correlation_matrix = np.corrcoef(X, rowvar=False)
+
+# Plot the lower triangle of the correlation matrix
+fig, ax = plt.subplots(figsize=(10, 8))
+mask = np.triu(np.ones_like(correlation_matrix, dtype=bool))
+sns.heatmap(
+    correlation_matrix,
+    mask=mask,
+    cmap="coolwarm",
+    annot=True,
+    fmt=".2f",
+    square=True,
+    cbar_kws={"shrink": 0.8},
+    ax=ax,
+)
+ax.set_title("Correlation Matrix")
+ax.set_yticklabels(feature_names, fontsize=10)
```

Comment:
could be more readable if the labels were displayed horizontally or in oblique.

Reply:
I added a rotation to the labels